core/services/ocr2/plugins/ocr2keeper/evm/log_filters.go

```diff
@@ -0,0 +1,102 @@
+package evm
+
+import (
+	"bytes"
+	"math/big"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/pkg/errors"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/logpoller"
+	"github.com/smartcontractkit/chainlink/v2/core/gethwrappers/generated/i_keeper_registry_master_wrapper_2_1"
+)
+
+const (
+	// logRetention is the amount of time to retain logs for.
+	// 5 minutes is the desired retention time for logs, but we add an extra 10 minutes buffer.
+	// TODO: TBD make this configurable or based on block time.
+	logRetention = (time.Minute * 5) + (time.Minute * 10)
+)
+
+// LogTriggerConfig is an alias for log trigger config.
+type LogTriggerConfig i_keeper_registry_master_wrapper_2_1.KeeperRegistryBase21LogTriggerConfig
+
+// logFilterManager manages log filters for upkeeps.
+type logFilterManager struct {
+	poller logpoller.LogPoller
+}
+
+func newLogFilterManager(poller logpoller.LogPoller) *logFilterManager {
+	return &logFilterManager{
+		poller: poller,
+	}
+}
+
+// Register creates a filter from the given upkeep and calls log pollet to register it
+func (lfm *logFilterManager) Register(upkeepID *big.Int, cfg LogTriggerConfig) error {
+	err := validateLogTriggerConfig(cfg)
+	if err != nil {
+		return errors.Wrap(err, "invalid log trigger config")
+	}
+	filter := newLogFilter(upkeepID, cfg)
+	// TODO: TBD remove old filter (if exist) for this upkeep
+	// _ = lfp.poller.UnregisterFilter(filterName(upkeepID), nil)
+	err = lfm.poller.RegisterFilter(filter)
+	return errors.Wrap(err, "failed to register upkeep filter")
+}
+
+// Unregister removes the filter for the given upkeepID
+func (lfm *logFilterManager) Unregister(upkeepID *big.Int) error {
+	err := lfm.poller.UnregisterFilter(filterName(upkeepID), nil)
+	return errors.Wrap(err, "failed to unregister upkeep filter")
+}
+
+// newLogFilter creates logpoller.Filter from the given upkeep config
+func newLogFilter(upkeepID *big.Int, cfg LogTriggerConfig) logpoller.Filter {
+	sigs := []common.Hash{
+		common.BytesToHash(cfg.Topic0[:]),
+	}
+	sigs = addFiltersBySelector(cfg.FilterSelector, sigs, cfg.Topic1[:], cfg.Topic2[:], cfg.Topic3[:])
```

Comment:
Rather than initialising the sigs slice outside of this function and then passing it as a param, does it make sense to instead pass topic0 as a param, and make the addFiltersBySelector function responsible for initialising, populating and returning the sigs slice?